source/util/FeatureRegistry.cpp

```diff
@@ -0,0 +1,90 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "FeatureRegistry.h"
+#include "../logging/LoggerFactory.h"
+
+using namespace Aws::Iot::DeviceClient;
+using namespace Aws::Iot::DeviceClient::Util;
+using namespace Aws::Iot::DeviceClient::Logging;
+
+constexpr char FeatureRegistry::TAG[];
+
+std::shared_ptr<Feature> FeatureRegistry::get(const std::string &name) const
```

Comment:
Is this used? I'm not seeing it used outside of your unit tests.

Reply:
I will be using get in dynamic reload

Reply:
I would probably just add it in your dynamic reload PR then. Adding a method you will need later that doesn't actually get used does not seem like a good idea. What if AppSec doesn't approve your project or it is never merged for some other reason?

Reply:
This method is prolly needed for testing.